src/api.cpp

```diff
@@ -199,7 +201,8 @@ void jit_record_end(JitBackend backend, uint32_t value) {
 
     // Set recording flag to previous value
     jit_set_flag(JitFlag::Recording, value & 0x80000000u);
-    value &= 0x7fffffff;
+    thread_state(backend)->recording_vcall = value & 0x40000000u;
```

Comment:
Is this representation (`recording_vcall`) sufficiently general? IMO we can have arbitrary levels of loops and vcalls. It would seem to me that we need some stack, or that the logic in `state.h` stores the previous value and updates it similar to other state flags in there.

Reply:
NVM, I will suggest a different approach in the main review comment.